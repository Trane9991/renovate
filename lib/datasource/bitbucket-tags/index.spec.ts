import { getPkgReleases } from '..';
import * as httpMock from '../../../test/httpMock';
import { id as datasource } from '.';

describe('datasource/bitbucket-tags', () => {
  beforeEach(() => {
    httpMock.reset();
    httpMock.setup();
  });
  describe('getReleases', () => {
    it('returns tags from bitbucket cloud', async () => {
      const body = {
        pagelen: 3,
        values: [
          {
            name: 'v1.0.0',
            target: {
              date: '2020-11-19T09:05:35+00:00',
            },
          },
          {
            name: 'v1.1.0',
            target: {},
          },
          {
            name: 'v1.1.1',
          },
        ],
        page: 1,
      };
      httpMock
        .scope('https://api.bitbucket.org')
        .get('/2.0/repositories/some/dep2/refs/tags')
        .reply(200, body);
      const res = await getPkgReleases({
        datasource,
        depName: 'some/dep2',
      });
      expect(res).toMatchSnapshot();
      expect(res.releases).toHaveLength(3);
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
  });
});
