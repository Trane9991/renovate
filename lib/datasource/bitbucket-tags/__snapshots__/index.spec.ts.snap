// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datasource/bitbucket-tags getReleases returns tags from bitbucket cloud 1`] = `
Object {
  "releases": Array [
    Object {
      "gitRef": "v1.0.0",
      "releaseTimestamp": "2020-11-19T09:05:35+00:00",
      "version": "v1.0.0",
    },
    Object {
      "gitRef": "v1.1.0",
      "version": "v1.1.0",
    },
    Object {
      "gitRef": "v1.1.1",
      "version": "v1.1.1",
    },
  ],
  "sourceUrl": "https://api.bitbucket.org/2.0/some/dep2",
}
`;

exports[`datasource/bitbucket-tags getReleases returns tags from bitbucket cloud 2`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "api.bitbucket.org",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://api.bitbucket.org/2.0/repositories/some/dep2/refs/tags",
  },
]
`;
